### PR TITLE
fix(#2210): log a warning if a final method is skipped when generating client proxies and subclasses

### LINKED ISSUE #2210: Log a warning if a final method is skipped when generating client proxies and subclasses
By default, CDI fails with a deployment problem if a type that should be proxyable declares a final method. ArC is more lenient and simply ignores final methods. We should log a warning message because invocation of a final method upon a client proxy/subclass may result in unpredictable behavior (interceptors are not applied, etc.).

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #2210 

@mkouba

### CODE REVIEW

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java [hunk 1]
```diff
@@ -148,7 +155,13 @@ static void addInterceptedMethodCandidates(BeanDeployment beanDeployment, ClassI
     }
 
     private static boolean skipForSubclass(MethodInfo method) {
-        if (Modifier.isStatic(method.flags()) || Modifier.isFinal(method.flags())) {
+        short flags = method.flags();
+        String className = method.declaringClass().name().toString();
+        if (Modifier.isFinal(flags) && !className.startsWith("java.")) {
+            LOGGER.warn(String.format("Method %s.%s() is final, skipped during generation of corresponding client proxy",
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: client proxy -> intercepted subclass

**Review by mkouba — CHANGES_REQUESTED** [review]
Otherwise looks good!

**Comment by machi1990** [comment]
> Otherwise looks good!

Thanks for the review. Comments addressed.

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java [hunk 2]
```diff
@@ -105,7 +106,13 @@ static void addDelegatingMethods(IndexView index, ClassInfo classInfo, Map<TypeV
     }
 
     private static boolean skipForClientProxy(MethodInfo method) {
-        if (Modifier.isStatic(method.flags()) || Modifier.isFinal(method.flags()) || Modifier.isPrivate(method.flags())) {
+        short flags = method.flags();
+        String className = method.declaringClass().name().toString();
+        if (Modifier.isFinal(flags) && !className.startsWith("java.")) {
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: Hm, is there actually a reason to skip the message for JDK classes?

Comment (machi1990) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: To keep the logs to a minimum user related stuffs.

Comment (mkouba) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java]: Ok makes sense. User can define a producer for a JDK class but it's probably reasonable.

**Review by mkouba — APPROVED** [review]

**Pull request merged**